### PR TITLE
 Set volumeType to tripleo if found empty 

### LINKED ISSUE #12084: Cluster update fails due to changed VolumeType
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Version 1.22.0-alpha.2 (git-345ef59498)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
kubernetesVersion: 1.21.3

**3. What cloud provider are you using?**
Red Hat OpenStack Platform 16.1

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
Deploy cluster with command
```
$ /home/stack/go/bin/kops create cluster \
  --cloud openstack \
  --name cluster.k8s.local \
  --state 'swift://kops' \
  --zones nova \
  --network-cidr 172.x.x.x/24 \
  --image bionic-root \
  --master-count=3 \
  --node-count=1 \
  --node-size med \
  --master-size med \
  --os-octavia=true \
  --topology private \
  --networking flannel \
  --os-ext-net public \
  --os-dns-servers=10.x.x.x \
  --ssh-public-key /home/stack/.ssh/id_rsa.pub \
  --bastion \
  --admin-access "0.0.0.0/0" \
  --master-volume-size 13 \
  --node-volume-size 13 \
  --container-runtime docker \
  --yes
```
Then update cluster with
```
$ kops update cluster --name cluster.k8s.local --yes --admin
```

**5. What happened after the commands executed?**
The cluster create command doesn't have `--etcd-storage-type` which means volumes created would be of the default type `tripleo`. 
Upon updating the cluster
```
(overcloud) [stack@undercloud-0 kops]$ /home/stack/go/bin/kops update cluster --name cluster.k8s.local --yes --admin 
I0801 01:41:18.300128  695714 apply_cluster.go:486] Gossip DNS: skipping DNS validation
I0801 01:41:20.077697  695714 executor.go:111] Tasks: 0 done / 111 total; 47 can run
W0801 01:41:20.644973  695714 executor.go:139] error running task "Volume/a.etcd-events.cluster.k8s.local" (9m59s remaining to succeed): Field cannot be changed: VolumeType
W0801 01:41:20.645020  695714 executor.go:139] error running task "Volume/a.etcd-main.cluster.k8s.local" (9m59s remaining to succeed): Field cannot be changed: VolumeType
I0801 01:41:20.645055  695714 executor.go:111] Tasks: 45 done / 111 total; 54 can run
W0801 01:41:22.115632  695714 executor.go:139] error running task "Volume/a.etcd-events.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
W0801 01:41:22.115677  695714 executor.go:139] error running task "Volume/a.etcd-main.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
I0801 01:41:22.115705  695714 executor.go:111] Tasks: 97 done / 111 total; 9 can run
W0801 01:41:22.327060  695714 executor.go:139] error running task "Volume/a.etcd-events.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
W0801 01:41:22.327119  695714 executor.go:139] error running task "Volume/a.etcd-main.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
I0801 01:41:22.327142  695714 executor.go:111] Tasks: 104 done / 111 total; 4 can run
W0801 01:41:22.403526  695714 executor.go:139] error running task "Volume/a.etcd-events.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
W0801 01:41:22.403561  695714 executor.go:139] error running task "Volume/a.etcd-main.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
I0801 01:41:22.403584  695714 executor.go:111] Tasks: 106 done / 111 total; 5 can run
W0801 01:41:22.833551  695714 bootstrapscript.go:154] exporting username and password. Consider using application credentials instead.
W0801 01:41:22.834843  695714 bootstrapscript.go:154] exporting username and password. Consider using application credentials instead.
W0801 01:41:23.019403  695714 bootstrapscript.go:154] exporting username and password. Consider using application credentials instead.
W0801 01:41:23.019937  695714 bootstrapscript.go:154] exporting username and password. Consider using application credentials instead.
W0801 01:41:23.020444  695714 executor.go:139] error running task "Volume/a.etcd-events.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
W0801 01:41:23.020465  695714 executor.go:139] error running task "Volume/a.etcd-main.cluster.k8s.local" (9m57s remaining to succeed): Field cannot be changed: VolumeType
I0801 01:41:23.020484  695714 executor.go:111] Tasks: 109 done / 111 total; 2 can run
W0801 01:41:23.078653  695714 executor.go:139] error running task "Volume/a.etcd-events.cluster.k8s.local" (9m56s remaining to succeed): Field cannot be changed: VolumeType
W0801 01:41:23.078689  695714 executor.go:139] error running task "Volume/a.etcd-main.cluster.k8s.local" (9m56s remaining to succeed): Field cannot be changed: VolumeType
```

**6. What did you expect to happen?**
Cluster update should've gone through successfully.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

```
(overcloud) [stack@undercloud-0 kops]$ /home/stack/go/bin/kops get --name cluster.k8s.local -o yaml                                                                                                                                                                                         
apiVersion: kops.k8s.io/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: "2021-07-30T09:51:19Z"
  name: cluster.k8s.local
spec:
  api: {}
  authorization:
    rbac: {}
  channel: stable
  cloudConfig:
    openstack:
      blockStorage:
        bs-version: v3
        ignore-volume-az: false
      monitor:
        delay: 1m
        maxRetries: 3
        timeout: 30s
      router:
        dnsServers: 10.19.42.41
        externalNetwork: public
  cloudProvider: openstack
  configBase: swift://kops/cluster.k8s.local
  containerRuntime: docker
  etcdClusters:
  - cpuRequest: 200m
    etcdMembers:
    - instanceGroup: master-nova
      name: a
    memoryRequest: 100Mi
    name: main
  - cpuRequest: 100m
    etcdMembers:
    - instanceGroup: master-nova
      name: a
    memoryRequest: 100Mi
    name: events
  iam:
    allowContainerRegistry: true
    legacy: false
  kubelet:
    anonymousAuth: false
  kubernetesApiAccess:
  - 0.0.0.0/0
  kubernetesVersion: 1.21.3
  masterPublicName: api.cluster.k8s.local
  networkCIDR: 172.22.0.0/24
  networking:
    flannel:
      backend: vxlan
  nonMasqueradeCIDR: 100.64.0.0/10
  sshAccess:
  - 0.0.0.0/0
  subnets:
  - cidr: 172.22.0.32/27
    name: nova
    type: Private
    zone: nova
  - cidr: 172.22.0.0/30
    name: utility-nova
    type: Utility
    zone: nova
  topology:
    dns:
      type: Public
    masters: private
    nodes: private

---

apiVersion: kops.k8s.io/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: "2021-07-30T09:51:20Z"
  labels:
    kops.k8s.io/cluster: cluster.k8s.local
  name: bastions
spec:
  image: bionic-root
  machineType: med
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: bastions
  role: Bastion
  subnets:
  - nova

---

apiVersion: kops.k8s.io/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: "2021-07-30T09:51:19Z"
  labels:
    kops.k8s.io/cluster: cluster.k8s.local
  name: master-nova
spec:
  image: bionic-root
  machineType: med
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-nova
  role: Master
  rootVolumeSize: 13
  subnets:
  - nova

---

apiVersion: kops.k8s.io/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: "2021-07-30T09:51:19Z"
  labels:
    kops.k8s.io/cluster: cluster.k8s.local
  name: nodes-nova
spec:
  image: bionic-root
  machineType: med
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: nodes-nova
  role: Node
  rootVolumeSize: 13
  subnets:
  - nova
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**
https://gist.github.com/yadneshk/bd5cf0370394cb1db584d06f51231291

**9. Anything else do we need to know?**
```
$ cat gist | grep expected
I0801 04:56:38.306974  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
I0801 04:56:38.320604  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
I0801 04:56:39.185063  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
I0801 04:56:39.216947  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
I0801 04:56:40.426898  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
I0801 04:56:40.427551  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
I0801 04:56:40.593285  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
I0801 04:56:40.594292  996012 changes.go:81] Field changed "VolumeType" actual="tripleo" expected=""
```
It seems that during cluster update the default volumetype tripleo is not recognized, however if a volumetype is passed in the cluster create command `(--etcd-storage-type tripleo)` this issue doesn't surface while cluster update.

I am working on the fix for this

### PR DESCRIPTION
**Pull request by yadneshk**
Fixes #12084

Signed-off-by: Yadnesh Kulkarni <ykulkarn@redhat.com>

### CODE REVIEW

**Comment by hakman** [comment]
@yadneshk Please remove the "Fixes #12084" from the commit message. This should fix the "invalid-commit-message" label.

File: pkg/model/master_volumes.go [hunk 1]
```diff
@@ -282,6 +282,9 @@ func (b *MasterVolumeBuilder) addGCEVolume(c *fi.ModelBuilderContext, name strin
 
 func (b *MasterVolumeBuilder) addOpenstackVolume(c *fi.ModelBuilderContext, name string, volumeSize int32, zone string, etcd kops.EtcdClusterSpec, m kops.EtcdMemberSpec, allMembers []string) error {
 	volumeType := fi.StringValue(m.VolumeType)
+	if volumeType == "" {
```
Comment (zetaab) [inline pkg/model/master_volumes.go:285]: I do not understand this change, why we should default to tripleo? That can exist in redhat openstack, but not in another openstacks.

**Review by zetaab — CHANGES_REQUESTED** [review]

File: pkg/model/master_volumes.go (hunk 1 above)
Comment (zetaab) [reply pkg/model/master_volumes.go:285]: instead if volumetype is empty, we should somehow detect that from openstack apis

Comment (yadneshk) [reply pkg/model/master_volumes.go:285]: @zetaab i didn't take into consideration that other openstack flavors won't default to tripleo. It should be fine to just pass volume type in the cluster create command. I'll close this PR. Thank you for reviewing.

**Pull request closed without merging**